src/main/java/javax/jmdns/impl/JmDNSImpl.java

```diff
@@ -1346,14 +1346,21 @@ void handleRecord(DNSRecord record, long now) {
             final boolean unique = newRecord.isUnique();
             final DNSRecord cachedRecord = (DNSRecord) this.getCache().getDNSEntry(newRecord);
             logger.debug("{} handle response cached record: {}", this.getName(), cachedRecord);
+            // RFC 6762, section 10.2 Announcements to Flush Outdated Cache Entries
+            // https://tools.ietf.org/html/rfc6762#section-10.2
+            // if (cache-flush a.k.a unique), remove all existing records matching these criterias :--
+            //     1. same name
+            //     2. same record type
+            //     3. same record class
+            //     4. record is older than 1 second.
             if (unique) {
                 for (DNSEntry entry : this.getCache().getDNSEntryList(newRecord.getKey())) {
                     if (    newRecord.getRecordType().equals(entry.getRecordType()) &&
                             newRecord.getRecordClass().equals(entry.getRecordClass()) &&
-                            (entry != cachedRecord)
+                            cachedRecord.getCreated() < (now - DNSConstants.FLUSH_RECORD_OLDER_THAN_1_SECOND*1000) 
```

Comment:
I would prefer a method here, like eg. `boolean isOlderThanOneSecond(DNSEntry)` to make it more readable and expressive.